posting/list.go

```diff
@@ -1087,10 +1087,22 @@ func (ro *rollupOutput) getRange(uid uint64) (uint64, uint64) {
 	return 1, math.MaxUint64
 }
 
+func shouldSplit(plist *pb.PostingList) (bool, error) {
+	r := roaring64.New()
```

Comment:
Do the cardinality check only when the first condition succeed.

Reply:
Done.